solr/core/src/java/org/apache/solr/update/SolrIndexWriter.java

```diff
@@ -176,14 +176,18 @@ private SolrIndexWriter(SolrCore core, String name, String path, Directory direc
     }
   }
 
+  public static void setCommitData(IndexWriter iw, long commitCommandVersion) {
```

Comment:
It appears this is now unused (sorta) I'd just remove it.  It's too internal to worry about back-compat.
I see one remaining caller; did you deliberately choose for that to not call the new version?

Reply:
The call from DirectUpdateHandler2.closeWriter() can provide a null param.
The call from SolrIndexSplitter.doSplit() actually I'm wondering now if it should set the custom user metadata by copying it from the latest commit of the core being split.

Reply:
The call from SolrIndexSplitter.doSplit() could set the commit metadata from the split command itself because it commits. I'll update this PR to also allow the (plugin writer) user to set commit metadata in the split command.